### PR TITLE
AMQ Streams intergration doc - initial

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: _artifacts/document-attributes-dm.adoc [hunk 1]
```diff
@@ -42,4 +42,6 @@
 :CENTRAL_CAPITAL_UNDER: DECISION_CENTRAL
 :CENTRAL_ONEWORD: decisioncentral
 
+:KAFKA: Red Hat AMQ Streams
```
Comment (sterobin) [inline _artifacts/document-attributes-dm.adoc:45]: @mramendi , we need to clarify some terminology here.  I already went through this for Kogito, and now that I see it here, I chatted with Ben Hardesty from AMQ to confirm. From my understanding on Kogito and now from Ben, you have it right for the platforms, but not for components of the platforms, such as messages, clusters, etc., which are still known as "Kafka messages", "Kafka clusters", even in enterprise. The exception is "AMQ Streams server" which is accurate I believe.

So to summarize, it's like this:

- AMQ Streams on RHEL roughly = Apache Kafka in community, really just based on it like you've said in the docs. So these attributes you have here are correct to use **for the platforms themselves**, but not for all components.
- AMQ streams on OpenShift = Strimzi in community, which is also based on Kafka but has that distinct community name. This doesn't seem to apply in this doc, but if it does, then the attributes will need to be revisited. I do see a link to both types of AMQ at the end of the doc as additional resources.
- Kafka messages, Kafka clusters, etc., are the correct terms for both enterprise and community. We shouldn't say AMQ Streams messages, AMQ Streams clusters, etc.
- AMQ Streams server is correct however, from what I understand.

So the only need change in this doc that I see is that all instances of `{KAFKA} messages` need to be changed to just `Kafka messages`, no attribute, to ensure it appears that way in both outputs. This includes all references used in the headings too. Only references to the platform and server should use the attribute, i.e., `{KAFKA} ifdef... based on Apache Kafka endif  is a streaming platform` is correct, but `{KAFKA} messages` isn't.

**Review by sterobin — COMMENTED** [review]
@mramendi Done, comments below. Ensure to expand any hidden comments.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+You can add a `KafkaPublishMessages` custom task to your process. This task sends {KAFKA} messages. It does not use the {KIE_SERVER} Kafka capability, so you can use this task in processes that do not run on a {KIE_SERVER}. However, it is more complicated to configure than other {KAFKA} integration options.
+
+.Procedure
+
+. In the {CENTRAL} administrative settings menu, as the administrative user, select *Custom Tasks Administration*.
+. Ensure that *KafkaPublishMessages* is set to *On*. 
+. In the {CENTRAL} main menu, select *Projects*.
+. Select your space and your project.
+. Select the *Settings* tab and then the *Custom Tasks* tab.
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc:12]: The real problem here is accessibility. "Settings" is in the top tabs and then "Custom tasks" is in the left tabs but accessbility guidelines disallow this. I think "select the Settings -> Custom Tasks" tab might work?

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+. In the *KafkaPublishMessages* line, click *Install*.
+. Enter the following information:
+** *Bootstrap Servers*: The host and port of the Kafka server, for example, `localhost:9092`. You can use a comma-separated list of multiple host:port pairs.
+** *Client ID*: An identifier string to pass to the server when making requests. The server uses this string for logging.
+** *Key Serializer class*: The class that provides the key serializer. Enter the standard serializer class name: `org.apache.kafka.common.serialization.StringSerializer`.
+** *Value Serializer class*: The class that provides the value serializer. Enter the standard serializer class name: `org.apache.kafka.common.serialization.StringSerializer`.
+. Select the *Deployments* tab and then the *Work item handlers* tab.
+. Click *Add work item handler*.
+. *I could not work out how to add dependencies and how to add the work item handler*
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc:21]: Intentional, will be deleted later - I just need to get Enrique's answer first

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/kieserver-kafka-emit-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* `id`: The unique identifier of the event
+* `type`: The type of the event (process, task, or case)
+* `source`: The event source as a URI
+* `time`: The timestamp of the event, by default in the https://tools.ietf.org/html/rfc3339[RFC3339] format
+* `data`: Information about the process, case, or task, presented in a JSON format
+
+.Procedure
+
+. To enable emitting {KAFKA} messages, complete one of the following steps:
+.. If you deployed the {KIE_SERVER} on {EAP} or another application server:
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/kieserver-kafka-emit-proc.adoc:29]: Unfortunately this results in the sub-steps being marked with bullets - AsciiDoctor limitation apparently, so I think the current formatting is the best I can do.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+When designing your business process in {CENTRAL}, you can create an event that receives {KAFKA} messages. 
+
+This event is triggered each time a message arrives in the configured topic. The message is expected to contain data that matches a predefined data object. The {PROCESS_ENGINE} parses the message and provides it as an output of the event.
+
+
+.Procedure
+
+. Open the project that contains your business process in {CENTRAL}.
+. Create a data object defining the data that the message will contain. For instructions about creating data objects, see {URL_DEVELOPING_PROCESS_SERVICES}#assembly-designing-business-processes[_{DESIGNING_BUSINESS_PROCESSES}_].
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc:12]: asked in chat too - I need the verified correct format for community doc cross-references

**Pull request merged**